### PR TITLE
storage: limit connection count through system variable

### PR DESCRIPTION
**Pull request by petrosagg**
### Motivation
This PR adds a `max_postgres_connections` and `max_kafka_connections` system vars that behave identically to the
`max_aws_privatelink_connections` flag.

### Checklist

- [ ] This PR has adequate test coverage / QA involvement has been duly considered.
- [ ] This PR has an associated up-to-date [design doc](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/design/README.md), is a design doc ([template](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/design/00000000_template.md)), or is sufficiently small to not require a design.
  
- [ ] If this PR evolves [an existing `$T ⇔ Proto$T` mapping](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/command-and-response-binary-encoding.md) (possibly in a backwards-incompatible way), then it is tagged with a `T-proto` label.
- [ ] If this PR will require changes to cloud orchestration or tests, there is a companion cloud PR to account for those changes that is tagged with the release-blocker label ([example](https://github.com/MaterializeInc/cloud/pull/5021)).
  
- [ ] This PR includes the following [user-facing behavior changes](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/guide-changes.md#what-changes-require-a-release-note):
  -

### CODE REVIEW

File: src/sql/src/session/vars.rs [hunk 1]
```diff
@@ -404,6 +404,21 @@ const TRANSACTION_ISOLATION: ServerVar<IsolationLevel> = ServerVar {
...
+    name: UncasedStr::new("max_kafka_connections"),
+    value: &0,
+    description:
+        "The maximum number of Kafka connections in the region, across all schemas (Materialize).",
+    internal: false,
+};
+
+pub const MAX_POSTGRES_CONNECTIONS: ServerVar<u32> = ServerVar {
+    name: UncasedStr::new("max_postgres_connections"),
+    value: &0,
```
Comment (andrewrodriguez-m) [inline src/sql/src/session/vars.rs:417]: do you want these to default to zero?

**Review by andrewrodriguez-m — APPROVED** [review]

File: src/sql/src/session/vars.rs (hunk 1 above)
Comment (petrosagg) [reply src/sql/src/session/vars.rs:417]: I'm not sure how we want to go about this. I think @benesch has a concrete plan in mind so let's see what he says

Comment (benesch) [reply src/sql/src/session/vars.rs:417]: I would default them to 1000 (to match `max_connections`) here, and then you won't need to do anything special in CI/tests to bump them to 9001; the default of 1000 should be sufficient.

We'll use LD to scope this down for sandbox trials.

**Review by benesch — APPROVED** [review]
It's so beautiful. Thank you!

**Pull request merged**